### PR TITLE
feat: first pass at WIP deployments overview

### PR DESCRIPTION
**Pull request by tunderwood**
Resolves GitHub issue: <link to GH issue>

Resolves Jira: [CDAAS-2507]

[CDAAS-2507]: https://armory.atlassian.net/browse/CDAAS-2507?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/deployment/overview.md [hunk 1]
```diff
@@ -8,3 +8,115 @@ categories: ["Canary Analysis", "Features", "Concepts"]
 tags: ["Deploy Strategy", "Canary", "Blue/Green", "Kubernetes"]
 ---
 
+
+
+## What is a Deployment
+Deployment is the encompassing construct that delivers your code to remote environments. A deployment
+can deliver software to a single environment or multiple environments either in sequence or in parallel
+depending on your [configuration]({{<ref "deployment/create-deploy-config" >}}).
```
Comment (parth-srivastav) [inline content/en/deployment/overview.md]:
```suggestion
depending on your [configuration]({{<ref "deployment/create-deploy-config" >}}).

You can define the CD-as-a-Service deployment configuration as a YAML file in a declarative manner. The configuration file can be stored within your source control, enabling code-like management. Deployments can be triggered from your CI system or by utilizing the Armory CLI. Currently, CD-as-a-Service requires a separate deployment configuration file for each application. However, you can deploy multiple Kubernetes deployment objects together as part of a single application. 
```

File: content/en/deployment/overview.md [hunk 2]
```diff
@@ -8,3 +8,128 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
```
Comment (parth-srivastav) [inline content/en/deployment/overview.md:16]: I believe we need a section here describing - how a deployment works. especially since we don't have dedicated pages right now for some of the stuff. I have a rough draft in the notion doc here
https://www.notion.so/RNA-Brief-Overview-239b993569b5485db7bc2d5b1fdfdc89

File: content/en/deployment/overview.md [hunk 3]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
 tags: ["Deploy Strategy", "Canary", "Blue/Green", "Kubernetes"]
 ---
 
+
+
+## What is a Deployment
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
## What a deployment is
```

File: content/en/deployment/overview.md [hunk 4]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+Deployment is the encompassing construct that delivers your code to remote environments. A deployment
+can deliver software to a single environment or multiple environments either in sequence or in parallel
+depending on your [configuration]({{<ref "deployment/create-deploy-config" >}}).
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
A _deployment_ encompasses the manifests, artifacts, configuration, and actions that deliver your code to remote environments. You can configure a deployment to deliver software to a single environment or multiple environments, either in sequence or in parallel depending on your [deployment configuration]({{<ref "deployment/create-deploy-config" >}}).
```

A construct doesn't perform actions - it's theoretical, a working hypothesis, a concept (yes I consulted [Merriam-Webster](https://www.merriam-webster.com/dictionary/construct) because "encompassing construct" just didn't sit will with me)

Style guide:
1. [Use simple and direct language](https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-simple-and-direct-language) - our readers may not be native English speakers
2. [Use active voice](https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-active-voice)

File: content/en/deployment/overview.md [hunk 5]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
+You can define the CD-as-a-Service deployment configuration as a YAML file in a declarative manner. The configuration file can be stored within your source control, enabling code-like management. Deployments can be triggered from your CI system or by utilizing the Armory CLI. Currently, CD-as-a-Service requires a separate deployment configuration file for each application. However, you can deploy multiple Kubernetes deployment objects together as part of a single application.
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
You define your CD-as-a-Service deployment configuration in a YAML file, which you store within your source control, enabling code-like management. You trigger deployments using the Armory CLI, either from your CI system or your workstation. Although CD-as-a-Service requires a separate deployment configuration file for each app, you can deploy multiple Kubernetes Deployment objects together as part of a single app. 
```

trigger deployment -> CLI (locally or from your CI); or use the GH action

Style:
1. [Use active voice](https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-active-voice)
2. "Currently..." [Avoid statements that may soon be out of date](https://docs.armory.io/contribute/create/guides/docs-style-guide/#avoid-statements-that-will-soon-be-out-of-date)
3. application -> app  (Google Developers Style Guide)

File: content/en/deployment/overview.md [hunk 6]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
+
+## What is a Deployment
+Deployment is the encompassing construct that delivers your code to remote environments. A deployment
+can deliver software to a single environment or multiple environments either in sequence or in parallel
+depending on your [configuration]({{<ref "deployment/create-deploy-config" >}}).
+
+You can define the CD-as-a-Service deployment configuration as a YAML file in a declarative manner. The configuration file can be stored within your source control, enabling code-like management. Deployments can be triggered from your CI system or by utilizing the Armory CLI. Currently, CD-as-a-Service requires a separate deployment configuration file for each application. However, you can deploy multiple Kubernetes deployment objects together as part of a single application. 
+
+## Defining a Deployment
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
## Defining your deployment
```
[Headings use sentence-style case](https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-sentence-capitalization-for-headings)

"Deployment" refers to a Kubernetes Deployment object

File: content/en/deployment/overview.md [hunk 7]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
+CD as a Service allows you to decoratively define your deployment configuration using a simple YAML file declaratively.
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
CD-as-a-Service enables you to declaratively define your deployment configuration in YAML file. 
```

allows -> enables   permission vs empowerment)

"simple" is a no go word because what one person thinks is simple might not be to another person. We could use "uncomplicated" but that falls into the same trap. Best to remove the adjective altogether.

File: content/en/deployment/overview.md [hunk 8]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+depending on your [configuration]({{<ref "deployment/create-deploy-config" >}}).
+
+You can define the CD-as-a-Service deployment configuration as a YAML file in a declarative manner. The configuration file can be stored within your source control, enabling code-like management. Deployments can be triggered from your CI system or by utilizing the Armory CLI. Currently, CD-as-a-Service requires a separate deployment configuration file for each application. However, you can deploy multiple Kubernetes deployment objects together as part of a single application. 
+
+## Defining a Deployment
+
+CD as a Service allows you to decoratively define your deployment configuration using a simple YAML file declaratively. 
+At its core you only need to provide only three pieces of information:
+
+* The environment/target you want to deploy to
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
* The target environment you want to deploy to
```

nit: pick a term rather than two with a slash
target/environme

File: content/en/deployment/overview.md [hunk 9]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+  - path: manifests/your-app.yaml # replace with the name of your app manifest
+    
+strategies:
+  <the-strategy-name>:
+    canary: # the strategy you want to use; current options are: canary, blue-green
+      ... # the configuration of your chosen strategy
+      
+```
+
+For more details, you can see our guide on how to [create your deployment configuration]({{<ref "deployment/create-deploy-config" >}}).
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
For deployment config file details, see {{< linkWithTitle "deployment/create-deploy-config.md" >}}.
```

[Avoid "we" and "our" ](https://docs.armory.io/contribute/create/guides/docs-style-guide/#avoid-using-we)

File: content/en/deployment/overview.md [hunk 10]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+strategies:
+  <the-strategy-name>:
+    canary: # the strategy you want to use; current options are: canary, blue-green
+      ... # the configuration of your chosen strategy
+      
+```
+
+For more details, you can see our guide on how to [create your deployment configuration]({{<ref "deployment/create-deploy-config" >}}). 
+
+## Elements of a Deployment
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
## Elements of a deployment
```

Or shorten to "Deployment elements"

File: content/en/deployment/overview.md [hunk 11]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
+### Deployment Targets/Environments
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
### Targets
```

Don't need to repeat "Deployment" since this is a subsection

File: content/en/deployment/overview.md [hunk 12]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+Within a deployment, you can define deployment targets that are equivalent to the environments you are
+deploying to.
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
Within a deployment, you define targets that are equivalent to the environments you are deploying to.
```

Removing "can" because that implies it's optional. You _can_ do it, but it's not a _must do_. 

I'm going to let slide ending the sentence with a preposition... but nobody but a grammar nerd cares. :)

File: content/en/deployment/overview.md [hunk 13]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
+For example, you may define staging, production-west, and production-east deployment targets.
```
Comment (aimeeu) [inline content/en/deployment/overview.md]: combine with previous paragraph - doesn't need to be its own paragraph

File: content/en/deployment/overview.md [hunk 14]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
+You can configure each deployment target/environment to use a [unique strategy]({{< ref "deployment/strategies/overview" >}})
+to orchestrate how your code is deployed and your traffic is routed.
```
Comment (aimeeu) [inline content/en/deployment/overview.md]: thank you for using the `ref` shortcode!!

```suggestion
You can configure each deployment target to use a [unique strategy]({{< ref "deployment/strategies/overview" >}}) that defines how to deploy your software and route traffic.
```

Use active voice

File: content/en/deployment/overview.md [hunk 15]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
+#### Deployment Target Constraints
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
#### Target constraints
```

File: content/en/deployment/overview.md [hunk 16]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+You can also configure your deployment targets to use constraints that prevent a deployment from beginning or completing until certain 
+conditions are met. For example, you can configure your deployment to wait for your code to be deployed to your staging
+environment before promoting that code to production.
+
+Constraints can be set as either `beforeDeployment` or `afterDeployment` depending on your use-case.
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
Constraints can be set as either `beforeDeployment` or `afterDeployment` depending on your use case. 
```

File: content/en/deployment/overview.md [hunk 17]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+to orchestrate how your code is deployed and your traffic is routed.
+
+#### Deployment Target Constraints
+You can also configure your deployment targets to use constraints that prevent a deployment from beginning or completing until certain 
+conditions are met. For example, you can configure your deployment to wait for your code to be deployed to your staging
+environment before promoting that code to production.
+
+Constraints can be set as either `beforeDeployment` or `afterDeployment` depending on your use-case. 
+
+Information on configuring constraints can be found in the [CD-as-a-Service reference docs]({{< ref "reference/deployment/config-file/targets#targetstargetnameconstraints" >}})
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
For info on configuring constraints, see the [CD-as-a-Service reference docs]({{< ref "reference/deployment/config-file/targets#targetstargetnameconstraints" >}})
```

active vs passive

File: content/en/deployment/overview.md [hunk 18]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+  target-2: #a user-defined name to reference this deployment target
+    account: <your-remote-network-agent-identifier> # the name you gave the RNA when you installed it in your prod cluster
+    namespace: <your-namespace-prod> # defined in namespace-prod.yaml
+    strategy: <the-name-of-the-strategy> # strategy will be defined below in the strategies block      
+    constraints:
+      dependsOn: ["target-1"] #A constraint indicating that the deployment to target-1 must complete successfully prior to beginning the deployment to target-2
+      beforeDeployment: # a list of constraints that must be satisfied before the deployment to the target can begin 
+        - pause: # a pause constraint that will pause the execution of the deployment until an authorized user gives an approval
+            untilApproved: true
+```
```
Comment (aimeeu) [inline content/en/deployment/overview.md]: turn the constraints into  bullet points and then the code as an example

```
CD-as-a-Service offers you multiple constraint options including: 

* `dependsOn`

   Use `dependsOn` to specify a target deployment that must successfully complete prior to starting this target's deployment. for 

* pause` for manual approval

   Use `pause` to pause a deployment for a set period of time or until an authorized user issues an approval. 


**Example**

  In the following example, you have two Kubernetes deployment targets: `target-1` and `target-2`.  You want deployment to `target-2` to depend on successful deployment to `target-1`, so you add a `dependsOn` constraint to `target-2`. Additionally, you want to manually approve deployment to `target-2`. In the `constraints` section, you add a `beforeDeployment` constraint with a `pause`.

```yaml
targets:
  target-1: #a user-defined name to reference this deployment target
    account: <your-remote-network-agent-identifier> # the name you gave the RNA when you installed it in your prod cluster
    namespace: <your-namespace-prod> # defined in namespace-prod.yaml
    strategy: <the-name-of-the-strategy> # strategy will be defined below in the strategies block      
  target-2: #a user-defined name to reference this deployment target
    account: <your-remote-network-agent-identifier> # the name you gave the RNA when you installed it in your prod cluster
    namespace: <your-namespace-prod> # defined in namespace-prod.yaml
    strategy: <the-name-of-the-strategy> # strategy will be defined below in the strategies block      
    constraints:
      dependsOn: ["target-1"] #A constraint indicating that the deployment to target-1 must complete successfully prior to beginning the deployment to target-2
      beforeDeployment: # a list of constraints that must be satisfied before the deployment to the target can begin 
        - pause: # a pause constraint that will pause the execution of the deployment until an authorized user gives an approval
            untilApproved: true
```

```

File: content/en/deployment/overview.md [hunk 19]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+### Deployment Manifests
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
### Manifests
```

File: content/en/deployment/overview.md [hunk 20]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
+Deploying your code with Armory CD-as-a-Service requires telling the CD-as-a-Service platform what artifacts should be 
+deployed. To accomplish this 
+you need to provide your Kubernetes manifests to Armory.
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
To deploy your software using CD-as-a-Service, you need tell CD-as-a-Service where to find your Kubernetes manifests.
```

[Use simple and direct language](https://docs.armory.io/contribute/create/guides/docs-style-guide/#use-simple-and-direct-language). Our readers have various English language skills.

File: content/en/deployment/overview.md [hunk 21]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+        - pause: # a pause constraint that will pause the execution of the deployment until an authorized user gives an approval
+            untilApproved: true
+```
+### Deployment Manifests
+
+Deploying your code with Armory CD-as-a-Service requires telling the CD-as-a-Service platform what artifacts should be 
+deployed. To accomplish this 
+you need to provide your Kubernetes manifests to Armory. 
+
+Manifests can be deployed to all environments or specific environments using a simple configuration:
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
You can deploy your manifests to all environments or specific environments using a basic configuration: 
```

Use active voice.
Avoid "simple"

File: content/en/deployment/overview.md [hunk 22]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+Manifests can be deployed to all environments or specific environments using a simple configuration: 
+
+```yaml
+manifests:
+  - path: manifests/your-manifest-1.yaml # this manifest will be deployed to all deployment targets defined in the targets block
+  - path: manifests/your-manifest-1.yaml # this manifest will be deployed to the 'target-2' target defined in the targets block
+    targets: ["target-2"]
+```
+             
+Manifest paths are relative to the directory from which you run the `armory deploy` command.
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
>Manifest paths are relative to the directory from which you run the `armory deploy` command.
```

Adding '>' which renders the sentence with "Note" styling.

File: content/en/deployment/overview.md [hunk 23]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+
+### Deployment Strategies
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
### Strategies 
```

File: content/en/deployment/overview.md [hunk 24]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+  - path: manifests/your-manifest-1.yaml # this manifest will be deployed to the 'target-2' target defined in the targets block
+    targets: ["target-2"]
+```
+             
+Manifest paths are relative to the directory from which you run the `armory deploy` command.
+
+### Deployment Strategies 
+A deployment strategy is the method by which your changes are deployed to a target. Strategies can use different techniques
+to allow for rapid rollback should a problem be discovered, minimizing the impact of potential issues to a small subset of users, 
+or can be optimized for speed.
```
Comment (aimeeu) [inline content/en/deployment/overview.md]:
```suggestion
A deployment strategy is the method by which CD-as-a-Service deploys your changes to a target. Strategies can use different techniques to allow for rapid rollback should a problem be discovered, minimizing the impact of potential issues to a small subset of users. You could also use a strategy optimized for speed. 
```

active voice

File: content/en/deployment/overview.md [hunk 25]
```diff
@@ -8,3 +8,130 @@ categories: ["Canary Analysis", "Features", "Concepts"]
...
+audience.
+
+For more details on Canary deployments and how to configure them, you can reference the [canary strategy documentation]({{< ref "deployment/strategies/canary" >}}) 
+              
+#### The Blue-Green Strategy
+In a blue-green deployment, two identical environments, blue and green, are maintained. The blue 
+environment represents the current production version, while the green environment represents the new version being 
+deployed. The new version is deployed and tested in the green environment, and once validated, traffic is routed to green instead of blue. This strategy minimizes downtime and provides a quick rollback option if issues arise.
+
+For more details on blue-green deployments and how to configure them, you can reference the [blue-green strategy documentation]({{< ref "deployment/strategies/blue-green" >}})
```
Comment (aimeeu) [inline content/en/deployment/overview.md]: Try not to use #### (which doesn't render in the right menu).
I'd use a bullet point list here. (I did the snippet in VSCode but can't guarantee the formatting pasted correctly)

```
CD-as-a-Service offers two distinct deployment strategies:

* Canary

   A canary deployment involves releasing a new software version to a small subset of users or systems while leaving the majority on the current version. This strategy allows for real-world testing and monitoring of the new version's performance and stability. If the canary users experience positive results, the new version can be gradually rolled out to a wider audience.

* Blue/Green

   In a blue/green deployment, you maintain two identical environments, blue and green. The blue environment represents the current production version, while the green environment represents the new version that is being deployed. The new version is deployed and tested in the green environment, and once validated, CD-as-a-Service routes traffic is to green instead of blue. This strategy minimizes downtime and provides a quick rollback option if issues arise.


For more details as to what deployment strategies are and how they differ, see ({{< linkWithTitle "deployment/strategies/overview.md" >}}).
```

I removed the direct links to each strategy page since it's probably better to drive people directly to the strategy overview page.

**Review by aimeeu — CHANGES_REQUESTED** [review]
Great draft! Lots of fabulous info.
Thank you for using the `ref` shortcode for site links!!! :)
I'd like to take a second pass after you've updated.

File: content/en/deployment/overview.md (hunk 4 above)
Comment (tunderwood) [reply content/en/deployment/overview.md]: Thanks @aimeeu !

I agree about the use of the term "encompassing construct". I'd actually love to get your input on how we should be talking about things because "deployment" has become an overloaded term, and I was trying to figure out how to explain that to a reader. 

It feels like we have three different definitions of deployment: 

- deployment - The "construct" I was trying to describe here. In other tools, this is called a "pipeline" but we've chosen not to use that terminology. The "deployment" in the armory doesn't actually perform any actions, it's a container that is made up of environment deployments. 
- environment deployment (my term, as I don't know that we actually have an agreed-upon term to describe this) - This is the thing that actually performs actions. At other companies, this is where I'd say something like, "I'm doing a production deployment" It is the set of actions taken to deploy code to a single remote environment. 
- Deployment - the k8s construct. Not really worried about this one, and I understand why we should be specific about "deployment" vs "Deployment". 

Basically, I was trying to explain the concept behind what Armory calls a "deployment" ( a set of 1 or more "environment deployments" configured in sequence or in parallel) vs. what I think most people think of when they read deployment (shipping code to an environment). 

I could absolutely be overthinking this but figured it was worth asking for your thoughts.

File: content/en/deployment/overview.md (hunk 2 above)
Comment (parth-srivastav) [reply content/en/deployment/overview.md:16]: I don't know how github PR reviews so I forgot to click the submit button. This doesn't need to be part of this PR, especially this late, and we can address it later.

**Review by aimeeu — APPROVED** [review]
